Python/Product/Analysis/Values/TypingTypeInfo.cs

```diff
@@ -435,13 +446,31 @@ sealed class TypingTypeInfoFinalizer {
                     if (!nameSet.Split(out IReadOnlyList<ConstantInfo> names, out var rest)) {
                         names = a.OfType<ConstantInfo>().ToArray();
                     }
-                    var name = names.Select(n => n.GetConstantValueAsString()).FirstOrDefault(n => !string.IsNullOrEmpty(n)) ?? "unnamed";
+                    name = names.Select(n => n.GetConstantValueAsString()).FirstOrDefault(n => !string.IsNullOrEmpty(n)) ?? "unnamed";
 
                     var p = new NamespaceProtocol(res, name);
-                    p.SetMember(node, unit, name, ToInstance(valueSet));
+                    var value = ToInstance(valueSet);
+                    p.SetMember(node, unit, name, value);
+                    tupleItem.Add(value);
                     res.AddProtocol(p);
+
+                    if (description.Any()) {
+                        description.Add(new KeyValuePair<string, string>(WellKnownRichDescriptionKinds.Comma, ", "));
+                    }
+                    description.AddRange(p.GetRichDescription());
                 }
+
+                res.AddProtocol(new TupleProtocol(res, tupleItem));
+            }
+
+            name = namedTupleName?.GetConstantValueAsString().FirstOrDefault() ?? "tuple";
```

Comment:
use `namedTupleName?.GetConstantValueAsString()?.FirstOrDefault()` because `null.FirstOrDefault()` causes an ANE.

Reply:
`GetConstantValueAsString()` will never return null, and if `namedTupleName` is null then the rest of that expression is skipped and it goes straight to the `??`.

Reply:
Ah, there's 2 GetConstantValueAsString and I looked at the wrong one...